sources/rules.h

```diff
@@ -115,8 +121,31 @@ struct od_rule
 	int                     client_max;
 	int                     log_debug;
 	od_list_t               link;
+    /* pause/resume related */
+    pthread_mutex_t lock;
+    od_rule_storage_state_t state;
 };
 
+static inline void
+od_rule_lock(od_rule_t *rule)
+{
+    pthread_mutex_lock(&rule->lock);
```

Comment:
We have our internal sleeplock.
It's there for inlining locks.

Actually, I'm not a big fan of bicycles, but, as long as it's already there, let's use it.

Reply:
I could not find any other lock. And other objects are locked with pthread_mutex_lock too. Could you give a link to certain file/line?